framework/base/ComponentTrait.php

```diff
@@ -0,0 +1,595 @@
+     * @return boolean whether the property can be read
+     * @see canSetProperty()
+     */
+    public function canGetProperty($name, $checkVars = true, $checkBehaviors = true)
+    {
+        if (method_exists($this, 'get' . $name) || $checkVars && property_exists($this, $name)) {
+            return true;
+        } elseif ($checkBehaviors) {
+            $this->ensureBehaviors();
+            foreach ($this->_behaviors as $behavior) {
+                if ($behavior->canGetProperty($name, $checkVars)) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Returns a value indicating whether a property can be set.
+     * A property can be written if:
+     *
+     * - the class has a setter method associated with the specified name
+     *   (in this case, property name is case-insensitive);
+     * - the class has a member variable with the specified name (when `$checkVars` is true);
+     * - an attached behavior has a writable property of the given name (when `$checkBehaviors` is true).
+     *
+     * @param string $name the property name
+     * @param boolean $checkVars whether to treat member variables as properties
+     * @param boolean $checkBehaviors whether to treat behaviors' properties as properties of this component
+     * @return boolean whether the property can be written
+     * @see canGetProperty()
+     */
+    public function canSetProperty($name, $checkVars = true, $checkBehaviors = true)
+    {
+        if (method_exists($this, 'set' . $name) || $checkVars && property_exists($this, $name)) {
+            return true;
+        } elseif ($checkBehaviors) {
+            $this->ensureBehaviors();
+            foreach ($this->_behaviors as $behavior) {
+                if ($behavior->canSetProperty($name, $checkVars)) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Returns a value indicating whether a method is defined.
+     * A method is defined if:
+     *
+     * - the class has a method with the specified name
+     * - an attached behavior has a method with the given name (when `$checkBehaviors` is true).
+     *
+     * @param string $name the property name
+     * @param boolean $checkBehaviors whether to treat behaviors' methods as methods of this component
+     * @return boolean whether the property is defined
+     */
+    public function hasMethod($name, $checkBehaviors = true)
+    {
+        if (method_exists($this, $name)) {
+            return true;
+        } elseif ($checkBehaviors) {
+            $this->ensureBehaviors();
+            foreach ($this->_behaviors as $behavior) {
+                if ($behavior->hasMethod($name)) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Returns a list of behaviors that this component should behave as.
```

Comment:
Using `@inheritdoc` in trait is incorrect, since traits doesn't handle any inheritance. It even break `yii2-apidoc` as far as I remember.